notmuch/db.c

```diff
@@ -102,6 +103,18 @@ notmuch_database_t *nm_db_do_open(const char *filename, bool writable, bool verb
   const notmuch_database_mode_t mode =
       writable ? NOTMUCH_DATABASE_MODE_READ_WRITE : NOTMUCH_DATABASE_MODE_READ_ONLY;
 
+  struct stat st1;
+  struct stat st2;
+  char buf[PATH_MAX];
+  if (stat(filename, &st1) != 0)
+    mutt_error(_("Can't stat %s: %s"), filename, strerror(errno));
+  else if (!S_ISDIR(st1.st_mode))
+    mutt_error(_("%s is not a directory"), filename);
+  if (stat(mutt_path_concat(buf, filename, ".notmuch", sizeof(buf)), &st2) != 0)
+    mutt_error(_("Can't stat %s: %s"), filename, strerror(errno));
+  else if(!S_ISDIR(st2.st_mode))
+    mutt_error(_("%s is not a directory"), buf);
+
```

Comment:
A few suggestions:

- If any of the checks fail, I think we should error message, then return
- Move the checks to the top of the function
- Do we need both `struct stat`?